common/elasticsearch/src/main/scala/uk/ac/wellcome/elasticsearch/IndexConfig.scala

```diff
@@ -93,3 +93,8 @@ trait IndexConfig {
   val canonicalId = lowercaseKeyword("canonicalId")
   val version = intField("version")
 }
+
+object NoStrictMapping extends IndexConfig {
```

Comment:
Would `DynamicMappingConfig` make more sense, as I think this is the use case?

I thought we were going to use `dynamic:false` though? So we could still specify mapping without them being indexed?